### DISCUSSION TITLE
open a file in the same window instead of starting a new one.

### DISCUSSION THREAD

**Discussion by bodnarlajos** [eclipse-theia/theia#13504] [Q&A]
Hi,
I use theia on linux and windows as well.
I would like to know how I can open a file from the file explorer in the running theia-ide window ?
on win10: I would like to use the "open with" for a file.

Thanks in advance!
Lajos

**Comment by msujew** [comment]
Hey @bodnarlajos,

this feature is closely related to https://github.com/eclipse-theia/theia/issues/10890. Theia doesn't support this by default. The proposed solution would be:

1. Enable the single instance lock
2. Send the info about the requested file to the electron window
3. Open the file in a new tab

This is obviously a very high level overview. The actual implementation requires a bit of effort.

**Reply by bodnarlajos** [reply]
Thank you for your answer.
How I can send the info to the electron window ? 
Could you send me a documentation about it ?

I was thinking about an extension what have a rest api and a small client app what can send the filepath ...
Can I run a nodejs script with theia-ide like: node somescript.js ?

Sorry for the lot of question but the ide seems very interesting and promising for me :)

**Reply by msujew** [reply]
The answer to these questions is a bit more complicated, maybe @tsmaeder can help here. I'm currently on vacation.

**Comment by tsmaeder** [comment]
We don't support opening files from the command line (https://github.com/eclipse-theia/theia/issues/11255). Adding this feature on the side seems awfully complicated. Why not propose a PR for the above issue? The relevant code lives in `ElectronMainApplication#handleMainCommand()`.
